sdl.cpp

```
@@ -0,0 +1,294 @@
+#include <string>
+#include <SDL.h>
+#include <SDL_image.h>
+#include <SWI-cpp2.h>
+
+PREDICATE(sdl_init_, 1)
+{
+  if (SDL_Init(A1.as_uint32_t()) < 0)
+  {
+    throw PlException(PlTerm_string(SDL_GetError()));
```

You might want to throw an exception in the standard form.
There's PlGeneralError, which is defined:
```cpp
PlGeneralError(PlTerm inside)
{ return PlException(PlCompound("error", PlTermv(inside, PlTerm_var())));
}
```

or `PlUnknownError()`, which takes a string.

---

done in [9ed33a9](https://github.com/kwon-young/sdlpl/pull/1/commits/9ed33a95866d09d8720eb19aea25d67035afeda9)